src/ThreeDSAuth.res

```diff
@@ -0,0 +1,94 @@
+open Utils
+let getKeyValue = (json, str) => {
+  json
+  ->Js.Dict.get(str)
+  ->Belt.Option.getWithDefault(Js.Dict.empty()->Js.Json.object_)
+  ->Js.Json.decodeString
+  ->Belt.Option.getWithDefault("")
+}
+
+@react.component
+let make = () => {
+  let (openModal, setOpenModal) = React.useState(_ => false)
+  let (loader, setloader) = React.useState(_ => true)
+
+  let logger = Recoil.useRecoilValueFromAtom(RecoilAtoms.loggerAtom)
+
+  React.useEffect0(() => {
+    handlePostMessage([("iframeMountedCallback", true->Js.Json.boolean)])
+    let handle = (ev: Window.event) => {
+      let json = ev.data->Js.Json.parseExn
+      let dict = json->Utils.getDictFromJson
+      if dict->Js.Dict.get("fullScreenIframeMounted")->Belt.Option.isSome {
+        let metadata = dict->getJsonObjectFromDict("metadata")
+        let metaDataDict =
+          metadata->Js.Json.decodeObject->Belt.Option.getWithDefault(Js.Dict.empty())
+        let paymentIntentId = metaDataDict->getString("paymentIntentId", "")
+        let headersDict =
+          metaDataDict
+          ->getJsonObjectFromDict("headers")
+          ->Js.Json.decodeObject
+          ->Belt.Option.getWithDefault(Js.Dict.empty())
+        let threeDsAuthoriseUrl =
+          metaDataDict
+          ->getJsonObjectFromDict("threeDSData")
+          ->Js.Json.decodeObject
+          ->Belt.Option.getWithDefault(Js.Dict.empty())
+          ->getString("three_ds_authorize_url", "")
```

Comment:
Could you please share the object response returned by backend in the PR description. Why are some keys snake_case (three_ds_authorize_url) and others camelCase (threeDSData)

Reply:
this is the POSTmessage we send which is in camelcase